nova/nova.go

```diff
@@ -744,3 +744,41 @@ func (c *Client) AttachVolume(serverId, volumeId, device string) (*VolumeAttachm
 	}
 	return &resp, nil
 }
+
+// DetachVolume detaches the volume with the given attachmentId from
+// the server with the given serverId.
+func (c *Client) DetachVolume(serverId, attachmentId string) error {
+	requestData := goosehttp.RequestData{}
+	url := fmt.Sprintf("%s/%s/%s/%s", apiServers, serverId, apiVolumeAttachments, attachmentId)
+	err := c.client.SendRequest(client.DELETE, "compute", url, &requestData)
+	if errors.IsNotFound(err) {
```

Comment:
The endpoint not existing? I don't know how common it is, but it's a definite possibility since this is supported only if the OpenStack instance supports volume extensions.